spec/requests/order/cancel_spec.rb

```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe "Order", type: :request do
+  describe "cancel" do
+    context "when the user does not login" do
+      it "will fail" do
+        cancel(1)
+        expect(response).to have_http_status(:found)
+      end
+    end
+
+    context "when the use login" do
+      it "will succeeded" do
+        login(user.email, user.password)
+        expect(response).to have_http_status(:redirect)
+        add_item_to_cart(book.id)
+        expect(response).to have_http_status(:ok)
+        payment
```

Comment:
I'm a bit confused here. How does payment work without mocking Stripe?

Reply:
Could we start with some unit tests for controller first, where you mock & stub Stripe (using https://github.com/stripe/stripe-mock or https://www.rubyguides.com/2018/10/rspec-mocks/)? We can use integration test after we we're more familiar with those concepts.